### PR TITLE
DOC: PR adds casting option's description to Glossary and `numpy.concatenate`.

### PR DESCRIPTION
**Pull request by MohitKumar020291**
Solves #20958. This PR resolves suggestions by @rossbar in #24057.

### CODE REVIEW

File: numpy/core/multiarray.py [hunk 1]
```diff
@@ -177,7 +177,9 @@ def concatenate(arrays, axis=None, out=None, *, dtype=None, casting=None):
 
     casting : {'no', 'equiv', 'safe', 'same_kind', 'unsafe'}, optional
         Controls what kind of data casting may occur. Defaults to 'same_kind'.
-
+        For more information
```
Comment (Mukulikaa) [inline numpy/core/multiarray.py:180]:
```suggestion
        For a description of the options, please see :term:`casting`.
```

File: numpy/core/multiarray.py [hunk 2]
```diff
@@ -177,7 +177,9 @@ def concatenate(arrays, axis=None, out=None, *, dtype=None, casting=None):
...
+        `<https://numpy.org/devdocs/glossary.html#glossary>`.
```
Comment (Mukulikaa) [inline numpy/core/multiarray.py:181]:
```suggestion
```

File: numpy/core/multiarray.py (hunk 1 above)
Comment (Mukulikaa) [reply numpy/core/multiarray.py:180]: NumPy docs can be interlinked with Sphinx directives.

File: doc/source/glossary.rst [hunk 3]
```diff
@@ -545,4 +545,26 @@ Glossary
          >>> x[0] = 3 # changing x changes y as well, since y is a view on x
          >>> y
          array([3, 2, 4])
-
+    
+   casting : Optional Controls what kind of data casting may occur during concatenation.
+
+       ``no``: No casting is allowed. The data types should not be cast at all.
+       Any mismatch in data types between the arrays being concatenated will
```
Comment (Mukulikaa) [inline doc/source/glossary.rst:552]:
```suggestion
       Any mismatch in data types between the arrays will
```

**Review by Mukulikaa — COMMENTED** [review]
Thanks for working on this, @MohitKumar020291. I requested a few stylistic changes. 
I am not very familiar with casting so I'll let someone else review the descriptions you have provided.

File: doc/source/glossary.rst [hunk 4]
```diff
@@ -545,4 +545,26 @@ Glossary
          >>> x[0] = 3 # changing x changes y as well, since y is a view on x
          >>> y
          array([3, 2, 4])
-
+    
+   casting : Optional Controls what kind of data casting may occur during concatenation.
```
[the remark this answers is not in the document]
Comment (MohitKumar020291) [reply doc/source/glossary.rst]: Thanks @Mukulikaa for correcting the mistake.

**Comment by MohitKumar020291** [comment]
@Mukulikaa Does I need to change something else? Please tell me. Thanks!

**Review by rossbar — APPROVED** [review]
I took the liberty of making a few final touchups:
 - Rebased on `main` (necessary to get the doc build working locally for whatever reason)
 - Moved the new casting term in the glossary to preserve alphabetical ordering
 - Reformatted the bullets so that the bulleted list renders correctly in the html docs.
 - Simplified the explanations of casting modes.

YMMV on the last change - IMO having a clear, concise description is better than a wall of text.

I'll let @Mukulikaa take another look to make sure I didn't break anything too badly, but this change LGTM - thanks @MohitKumar020291 !

**Comment by mattip** [comment]
Cool. The glossary reference as rendered in [concatenate](https://output.circle-artifacts.com/output/job/018de5fc-2623-435f-b8ba-39c7e77bad3d/artifacts/0/doc/build/html/reference/generated/numpy.concatenate.html#numpy-concatenate) looks nice, perhaps we can use that elsewhere in a follow up PR.

**Comment by MohitKumar020291** [comment]
@rossbar Thanks for some important changes and review.

**Comment by Mukulikaa** [comment]
Thanks, @MohitKumar020291 and @rossbar. It looks good- [casting](https://output.circle-artifacts.com/output/job/018de5fc-2623-435f-b8ba-39c7e77bad3d/artifacts/0/doc/build/html/glossary.html#term-casting).

**Comment by MohitKumar020291** [comment]
Thanks a lot to @Mukulikaa and @rossbar. You both made PR look good.

**Pull request merged**